core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -114,9 +117,35 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
   private def sortForExecutiveSummary(appsSumDesc: Seq[QualificationSummaryInfo],
       order: String): Seq[EstimatedSummaryInfo] = {
     if (QualificationArgs.isOrderAsc(order)) {
-      appsSumDesc.reverse.map(_.estimatedInfo)
+      appsSumDesc.reverse.map(sum =>
+        EstimatedSummaryInfo(
+          sum.estimatedInfo, sum.estimatedFrequency.getOrElse(DEFAULT_JOB_FREQUENCY)))
     } else {
-      appsSumDesc.map(_.estimatedInfo)
+      appsSumDesc.map(sum => EstimatedSummaryInfo(
+        sum.estimatedInfo, sum.estimatedFrequency.getOrElse(DEFAULT_JOB_FREQUENCY)))
+    }
+  }
+
+  // Estimate app frequency based off of all applications in this run, unit jobs per month
+  private def estimateAppFrequency(
+    appsSum: Seq[QualificationSummaryInfo]): Seq[QualificationSummaryInfo] = {
+    val appFrequency = scala.collection.mutable.Map[String, Double]()
+    var windowStart: Long = Long.MaxValue
+    var windowEnd: Long = Long.MinValue
+
+    appsSum.foreach { sum =>
+      appFrequency += (sum.appName -> (1.0 + appFrequency.getOrElse(sum.appName, 0.0)))
+      windowStart = Math.min(sum.startTime, windowStart)
```

Comment:
So this is assuming uniform distribution right?  Meaning what if I have apps for the last year say Jan - Dec and I started running an app in July once a month.  This estimation is not going to be correct as it should look at the timeframe for each app.  ie May-Dec and should come out with once per month, vs coming out once every 2 months.   Seems like to be correct you would track the times for that specific application.

Not sure exact requirement for this implementation, if that is ok and we document or we want to change it? @mattahrens

Reply:
ok lets just make sure behavior is documented